doc/globals.rst

```diff
@@ -0,0 +1,31 @@
+==========
+Singletons
```

Comment:
the file is  called 'globals' the the title is 'singletons'... we should use just one name...

you write the docs as if we would have many globals, not sure that would be the case. maybe we could expose the Reporter or some other internal stuff. But those are already documented elsewhere...

what about have this section of docs being specific to 'db backend' or something like that.

Another thing i just realized is that we are calling it 'dep_manager' as it is original function on doit itself.Wondering if we should use a more generic name. In practice users could use this to store some data. like a a task that stores and use a data itself in future runs. for  this use-case the name dep_manager is completely innapropriate.

what do you think?

Reply:
I'll adapt the section title to match file and module name.

> you write the docs as if we would have many globals

Correct, this way, it will be easy to extend it when needed. I certainly would not add anything here just to have more than one attribute.

Regarding the names: I surely stumbled upon the names initially as well. There is a class `Dependency` that is called `dep_manager` wherever it is used and in fact it is actually the doit database...

But I got over it 😃  If you want to rename it and clean this up, I'd ask you to do this in a separate PR, if you don't mind. I maybe would have called things differently here and there, but that is really a personal thing...